apps/sms/services/test/unit/moz_mobile_message/moz_mobile_message_shim_test.js

```diff
@@ -0,0 +1,375 @@
+/*global MozMobileMessageShim,
+         MocksHelper,
+         bridge
+*/
+
+'use strict';
+
+require('/services/test/unit/mock_bridge.js');
+require('/services/test/unit/mock_navigatormoz_sms.js');
+require('/views/shared/js/utils.js');
+
+const EVENTS = ['received', 'sending', 'sent', 'failed', 'deleted',
+                'readsuccess', 'deliverysuccess'];
+
+const METHODS = ['getMessage', 'retrieveMMS', 'send', 'sendMMS',
+                 'delete', 'markMessageRead', 'getSegmentInfoForText'];
+
+const STREAMS = ['getThreads', 'getMessages'];
+
+var MocksHelperForAttachment = new MocksHelper([
+  'bridge'
+]).init();
+
+suite('MozMobileMessageShim >', function() {
+  var realMozMobileMessage;
+  var serviceStub;
+
+  MocksHelperForAttachment.attachTestHelpers();
+
+  suiteSetup(function(done) {
+    realMozMobileMessage = navigator.mozMobileMessage;
+    navigator.mozMobileMessage = MockNavigatormozMobileMessage;
+
+    serviceStub = sinon.stub({
+      method: () => {},
+      stream: () => {},
+      broadcast: () => {}
+    });
+
+    sinon.stub(bridge, 'service').returns(serviceStub);
+    sinon.stub(MockNavigatormozMobileMessage, 'addEventListener');
+
+    require('/services/js/moz_mobile_message/moz_mobile_message_shim.js', done);
+  });
+
+  suiteTeardown(function() {
+    navigator.mozMobileMessage = realMozMobileMessage;
+  });
+
+  test('bridge service is correctly initialized', function() {
+    sinon.assert.calledOnce(bridge.service);
+    sinon.assert.calledWith(bridge.service, 'mozMobileMessageShim');
+  });
+
+  test('event listener is correctly initialized', function() {
+    EVENTS.forEach((event) => {
+      sinon.assert.calledWith(
+        MockNavigatormozMobileMessage.addEventListener,
+        event,
+        MozMobileMessageShim[Utils.camelCase(`on-${event}`)]
+      );
+    });
+  });
+
+  test('method mapping is correctly initialized', function() {
+    METHODS.forEach((shimMethod) => {
+      sinon.assert.calledWith(
+        serviceStub.method,
+        shimMethod,
+        MozMobileMessageShim[shimMethod]
+      );
+    });
+  });
+
+  test('stream mapping is correctly initialized', function() {
+    STREAMS.forEach((shimStream) => {
+      sinon.assert.calledWith(
+        serviceStub.stream,
+        shimStream,
+        MozMobileMessageShim[shimStream]
+      );
+    });
+  });
+
+  suite('event broadcast', function() {
+    var onEvent;
+
+    setup(function() {
+      onEvent = {
+        message: {},
+        deletedThreadIds: [1, 2, 3]
+      };
+    });
+
+    test('onSending', function() {
+      MozMobileMessageShim.onSending(onEvent);
```

Comment:
It's not silly and I thought the same previously, but for the method part I have some concern in https://github.com/mozilla-b2g/gaia/pull/30485#discussion_r32289343 and that make me prefer to keep the test separated.